recipe_scrapers/joshuaweissman.py

```diff
@@ -0,0 +1,89 @@
+# mypy: allow-untyped-defs
+
+# from pprint import pprint
+from ._abstract import AbstractScraper
+from ._grouping_utils import group_ingredients
+from ._utils import get_minutes, get_yields, normalize_string
+
+
+class JoshuaWeissman(AbstractScraper):
+    @classmethod
+    def host(cls):
+        return "joshuaweissman.com"
+
+    def author(self):
+        return self.schema.author()
+
+    def title(self):
+        return self.schema.title()
+
+    def category(self):
+        return self.schema.category()
+
+    def total_time(self):
+        total = self.soup.find(
+            lambda tag: tag.name == "span" and "Total Time" in tag.get_text()
+        )
+        if total:
+            return get_minutes(total)
+        prep = self.soup.find(
+            lambda tag: tag.name == "span" and "Prep Time" in tag.get_text()
+        )
+        if prep:
+            prep = get_minutes(prep)
+        cook = self.soup.find(
+            lambda tag: tag.name == "span" and "Cook Time" in tag.get_text()
+        )
+        if cook:
+            cook = get_minutes(cook)
+        return prep + cook
+
+    def yields(self):
+        return get_yields(
+            self.soup.find(
+                lambda tag: tag.name == "span" and "Serving Size" in tag.get_text()
+            )
+        )
+
+    def image(self):
+        return self.schema.image()
+
+    def ingredients(self):
+        ingredients = [
+            li.get_text() for ul in self.soup.findAll("ul") for li in ul.findAll("li")
+        ]
+
+        ingredients = [
+            normalize_string(ingredient) for ingredient in ingredients if ingredient
+        ]
+
+        return ingredients
+
+    def ingredient_groups(self):
+        # This is the css for it but web site has weird structuring so it doesn't work properly
+        return group_ingredients(
+            self.ingredients(),
+            self.soup,
+            'span[style="font-weight:700"]',
```

Comment:
Changed this as mentioned, and works overall quite well. The main case that this doesn't work is when the instructions have a section header with the exact same name as an ingredient group, which i only have found one recipe that does it. This does cause the `group_ingredients` to be overwritten with empty lists. I looked into some more options for it, but this seems overall to be the most robust way still, even though it doesn't work rarely

Reply:
Seems reasonable, thanks for investigating this!